types/test/XMLHttpRequest-compat.test-d.ts

```diff
@@ -0,0 +1,7 @@
+import { expectError } from 'tsd'
+import { newMockXhr } from '../'
+
+const xhr = newMockXhr()
+
+//can be used to validate structural compliance with XMLHttpRequest interface
+expectError<XMLHttpRequest>(xhr) //not yet compatible
```

Comment:
That's an interesting test! I don't think it will ever become `expectType` but I'd still like to keep it if only for documentation purposes.

Reply:
I considered this a good usage example, so the doc purpose was pretty much the intent behind it.